sdk/Lusid.Sdk/Utilities/ApiRetryHandler.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using Polly;
+using RestSharp;
+
+namespace Lusid.Sdk.Utilities
+{
+    /// <summary>
+    /// Class used to define API error retry rules for all API calls.
+    /// </summary>
+    public static class ApiRetryHandler
+    {
+        /// <summary>
+        /// Number of max retry attempts
+        /// </summary>
+        public const int MaxRetryAttempts = 3;
+
+        /// <summary>
+        /// Get the internal exception condition on which to retry.
+        /// </summary>
+        /// <param name="restResponse">Response object that comes from the API Client</param>
+        /// <returns></returns>
+        private static bool GetInternalExceptionRetryCondition(IRestResponse restResponse)
+        {
+            return restResponse.ErrorException != null || restResponse.StatusCode == 0;
+        }
+
+        /// <summary>
+        /// Define Polly retry policy for synchronous API calls. Handles internal SDK exceptions only.
+        /// Use .Wrap() method to combine this policy with your other custom policies
+        /// </summary>
+        public static Policy<IRestResponse> GetInternalExceptionRetryPolicy() =>
+            Policy
+                .Handle<SystemException>()
+                .OrResult<IRestResponse>(GetInternalExceptionRetryCondition)
+                .Retry(
+                    MaxRetryAttempts,
+                    onRetry: (result, retryCount, context) =>
+                    {
+                        Console.WriteLine("An internal exception has occurred. Retrying. " +
+                                          $"Retry attempt: {retryCount}. Max retry attempts: {MaxRetryAttempts}");
```

Comment:
This could become quite chatty. Is there a way to implement this using a logging library like you can do in Python where the user defines if they want to display the logs or not?

Reply:
This will occur only once in roughly 50k API calls the last time we ran a test in the pipeline. So quite rare. It should work after a single retry.

I'll see if we can wrap these in any logging libraries

Also, we want to make sure that we do have these logs as warnings and an indication that the underlying issue of the SDK still exists. Because retrying is just kind of putting a plaster on it - what do you think?

Reply:
Also, if we were to implement configurable logging, it should probably be a separate PR

Reply:
Fair enough, let's handle this separately.